### PR TITLE
Add a bunch of debugging to .well-known IS validation

### PR DESCRIPTION
**Pull request by turt2live**

### CODE REVIEW

File: src/autodiscovery.js [hunk 1]
```diff
@@ -296,6 +298,7 @@ export class AutoDiscovery {
                 wellknown["m.identity_server"]["base_url"],
             );
             if (!isUrl) {
+                logger.log("[Debug] First error: isUrl is falsey");
```
Comment (uhoreg) [inline src/autodiscovery.js]: This seems unnecessary given that there's a `logger.error`statement right below?

File: src/autodiscovery.js [hunk 2]
```diff
@@ -308,11 +311,20 @@ export class AutoDiscovery {
                 `${isUrl}/_matrix/identity/api/v1`,
             );
             if (!isResponse || !isResponse.raw || isResponse.action !== "SUCCESS") {
+                logger.log("[Debug] Second error: response from IS failed");
```
Comment (uhoreg) [inline src/autodiscovery.js]: Also possible unnecessary.

File: src/autodiscovery.js (hunk 1 above)
Comment (turt2live) [reply src/autodiscovery.js]: in theory, yes. There's about 2-3 places where we say `Invalid base_url for m.identity_server` on the error channel though :(

File: src/autodiscovery.js (hunk 2 above)
Comment (turt2live) [reply src/autodiscovery.js]: this one is, yes. I plan on just reverting this whole commit once I don't need it anymore, and am feeling a bit lazy to fix this

**Pull request merged**